bentoml/_internal/server/__init__.py

```diff
@@ -107,12 +146,35 @@ def serve_development(
         watchers,
         sockets=list(circus_socket_map.values()),
     )
+    tracking_thread, stop_thread_event = scheduled_track(
+        current_pid,
+        event_properties=get_scheduled_event_properties(
+            production=False,
+            bento_identifier=bento_identifier,
+            serve_info=serve_info,
+        ),
+    )
     _ensure_prometheus_dir()
 
-    arbiter.start(
-        cb=lambda _: logger.info(  # type: ignore
-            f'Starting development BentoServer from "{bento_identifier}" '
-            f"running on http://{host}:{port} (Press CTRL+C to quit)"
+    try:
+        tracking_thread.start()
+        arbiter.start(
+            cb=lambda _: logger.info(  # type: ignore
+                f'Starting development BentoServer from "{bento_identifier}" '
+                f"running on http://{host}:{port} (Press CTRL+C to quit)"
+            ),
+        )
+    finally:
+        stop_thread_event.set()
```

Comment:
There're two things here:
* we told the thread to stop but don't wait for it. suggestion: use thread.join() after this line.
* too many lines here in the caller. I think we may shield this logic in a custom context manager, like this 
```
with schedule_track(**options).start():
  arbiter.start(...)
```

Reply:
- thread cannot join since this is not a poolexecutor.
- thread is not a context manager since it doesn't have `__enter__`, so cannot start with a `with` block.

Reply:
@aarnphm I mean, to write your custom context manager, making use of try...except and event

Reply:
```python
# utils/tracking.py

@contextlib.contextmanager
def server_tracking(...):
    tracking_thread, stop_thread_event = scheduled_track(
        current_pid,
        event_properties=get_scheduled_event_properties(
            production=False,
            bento_identifier=bento_identifier,
            serve_info=serve_info,
        ),
    )
    try:
        tracking_thread.start()
        yield
    finally:
        stop_thread_event.set()
        tracking_thread.join()

# caller
with server_tracking():
        arbiter.start(
            cb=lambda _: logger.info(  # type: ignore
                f'Starting development BentoServer from "{bento_identifier}" '
                f"running on http://{host}:{port} (Press CTRL+C to quit)"
            ),
        )
```

Reply:
@aarnphm https://docs.python.org/3/library/threading.html#threading.Thread.join